internal/dataapi/listener.go

```diff
@@ -71,10 +70,10 @@ func (lis *Listener) Run(ctx context.Context) {
 	}
 
 	srv := &http.Server{
-		Handler:  srvHandler,
+		Handler:  lis.srv.ConnInfoMiddleware(srvHandler),
 		ErrorLog: slog.NewLogLogger(lis.opts.L.Handler(), slog.LevelError),
 		BaseContext: func(net.Listener) context.Context {
-			return conninfo.Ctx(ctx, conninfo.New())
+			return ctx
```

Comment:
Previously a `conninfo.New()` at the startup and all client shared the same conninfo. The change in this PR creates a `conninfo.New()` for each dataapi http request and untracks `conninfo` at the end of the request.